### ISSUE TITLE
Inline nano_id implementation?

### ISSUE THREAD

**Issue by coezbek** [schpet/cool_id#9]
Thank you for designing this Gem. I think you nailed it how it should be done.

Two questions:
1. could you imagine inlining the nanoid generation into cool_id? I have looked at the implementation of nanoid on https://github.com/radeno/nanoid.rb/ and am not convinced of the code quality.

    Since nanoid is essentially only the following code, it should be easy to add:

    ```
    alphabet = "-_0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz".chars
    digits = 21
    SecureRandom.alphanumeric(digits, chars: alphabet)
    ```

2. The documentation currently is missing a couple aspects:

    - Can prefix be undefined? What happens if the base class setup is used, but a descendant doesn't define a prefix?
    - When generating IDs there still can be a race-condition if the same ID is claimed, right?
    - It would be nice to add the migration needed for "using a different id field", I guess `rails g migration AddCoolIdToTable table cool_id:string:uniq:index` or `rails g migration AddCoolIdToTable table 'cool_id:string!:uniq:index' if on Rails 8

**Comment by FrankFang** [comment]
>  Can prefix be undefined?

No, See https://github.com/schpet/cool_id/blob/3a30e7554a51d5fe15b6930d52e65273e86a469d/lib/cool_id.rb#L186

**Comment by FrankFang** [comment]
> When generating IDs there still can be a race-condition if the same ID is claimed, right?

Yes, but the probability of this happening is extremely low.
https://github.com/schpet/cool_id/blob/3a30e7554a51d5fe15b6930d52e65273e86a469d/lib/cool_id.rb#L74

**Comment by schpet** [comment]
@coezbek thanks for the kind words! and i didn't realize SecureRandom could do that... that seems a lot better. i'll try and get a PR out with that. 

as for docs @FrankFang's comments look accurate to me, but i don't use the gem in this way so i can't really say for sure. i'm open to any changes that clarify those cases in the docs, or provide examples around migrations.
